### PR TITLE
Allow task exclusion in included builds

### PR DESCRIPTION
**Pull request by donat**
This PR allows users to exclude tasks from the included builds.

The implementation only changes how the filters are applied to the task graph. The existing implementation defines all filters (i.e. `-x` arguments on the command line) on the root build's graph. This PR inspects the filter's path and if it points to an included build, then the filter is assigned to the target build's task graph.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

File: subprojects/core/src/main/java/org/gradle/execution/ExcludedTaskFilteringBuildConfigurationAction.java [hunk 1]
```diff
@@ -38,11 +39,14 @@ public void configure(BuildExecutionContext context) {
...
-            final Set<Spec<Task>> filters = new HashSet<Spec<Task>>();
+            Multimap<GradleInternal, Spec<Task>> filters = MultimapBuilder.linkedHashKeys().hashSetValues().build();
             for (String taskName : excludedTaskNames) {
-                filters.add(taskSelector.getFilter(taskName));
+                TaskFilter filter = taskSelector.getFilter(taskName);
+                filters.put(filter.getGradle(), filter.getFilterSpec());
+            }
+            for (GradleInternal gradleInternal : filters.keySet()) {
+                gradleInternal.getTaskGraph().useFilter(Specs.intersect(filters.get(gradleInternal)));
             }
```
Comment (big-guy) [inline subprojects/core/src/main/java/org/gradle/execution/ExcludedTaskFilteringBuildConfigurationAction.java:49]: I think this looks a little weird for two reasons:
- We're pushing things down into each build. 
- We're creating separate filter lists for each build. 

Could we instead add a service to the Build Tree scope that represents the task filter and then have each DefaultTaskExecutionGraph use that service as the source of filters? Then here, we wouldn't care about any particular Gradle instance, we'd just populate the "task filter service". That would remove the need for `useFilter` on the task graph.

**Review by big-guy — CHANGES_REQUESTED** [review]

**Comment by donat** [comment]
Superseded by #15141

**Pull request closed without merging**